src/LevelDBIntegration.js

```diff
@@ -0,0 +1,53 @@
+import { ipcMain } from 'electron';
+import level from 'level';
+
+// global store for db connections
+const dbStore = {};
+
+/**
+ * This method will connect to a leveldb database and store a reference to dbStore.
+ * Note that leveldb only allows a single connection at a time so you need to
+ * explicitly close the connection to allow other processes to access it.
+ */
+function connectToLevelDb(path, createIfMissing = false, valueEncoding = 'json', ipcEvent = null) {
```

Comment:
@emanlodovice We can convert `createIfMissing`, `valueEncoding`, and `ipcEvent` into an object parameter containing those fields, instead of using positional parameters. That way we can omit some fields if we just want to use their default values. Also makes it easier to add more fields in the future.

Note that passing `undefined` will make the field use its default value, so there's also no more need to do `params.createIfMissing || false` etc. in `enableDBConnect` below.